docker/linux/base/build/Dockerfile

```diff
@@ -98,18 +98,18 @@ RUN cd /tmp \
     && rm clang-${CLANG_VER}.tgz
 
 # === Install Clang 16 toolchain for all Linux-hosted builds
-ARG CLANG_16_VER
+ARG CLANG_17_VER
```

Comment:
Perhaps we should remove clang major version from the variable names here ? Otherwise it's just churn in the future.